### ISSUE TITLE
Internal error related to instances

### ISSUE THREAD

**Issue by nad** [agda/agda#1959]
The following code triggers internal errors (using 0b06e89ce446194f70871d5481b2e2c3318464d8):

```agda
data _≡_ {A : Set} (x : A) : A → Set where
  refl : x ≡ x

cong : {A B : Set} (f : A → B) {x y : A} → x ≡ y → f x ≡ f y
cong f refl = refl

id : {A : Set} (x y : A) → x ≡ y → x ≡ y
id _ _ x≡y = x≡y

record R₁ (M : Set → Set) : Set₁ where
  field
    f : ∀ {A} → A → M A

open R₁ ⦃ ... ⦄ public

record R₂ (M : Set → Set) : Set₁ where
  field r₁ : R₁ M
  postulate
    lemma : ∀ {A} (x : A) → R₁.f r₁ x ≡ R₁.f r₁ x

open R₂ ⦃ ... ⦄ public

record Id (A : Set) : Set where
  constructor wrap
  field
    run : A

record App (M : Set → Set) (A : Set) : Set where
  constructor wrap
  field
    run : M A

instance

  postulate
    i₁ : {M : Set → Set} ⦃ r₁ : R₁ M ⦄ → R₁ (App M)

  i₂ : R₂ Id
  Id.run (R₁.f (R₂.r₁ i₂) x) = x

-- An internal error has occurred. Please report this as a bug.
-- Location of the error: src/full/Agda/TypeChecking/CompiledClause/Match.hs:191

internal-error₁ :
  ∀ {M A} (m : R₂ M) (a : R₂ (App M)) (x : A) →
  wrap (R₁.f (R₂.r₁ m) x) ≡ R₁.f (R₂.r₁ a) x
internal-error₁ _ _ x =
  cong wrap (id (f x) (f x) (lemma _))

-- An internal error has occurred. Please report this as a bug.
-- Location of the error: src/full/Agda/TypeChecking/Records.hs:479

internal-error₂ :
  ∀ {M A} (m : R₂ M) (a : R₂ (App M)) (x : A) →
  wrap (R₁.f (R₂.r₁ m) x) ≡ R₁.f (R₂.r₁ a) x
internal-error₂ _ _ x =
  id (wrap (f x)) (wrap (f x)) (lemma _)
```

If Agda 2.4.2.5 is used, then `internal-error₁` seems to make Agda loop, while `internal-error₂` triggers the following error message:

```
No variable of type R₂ (_M_60 .m .a x) was found in scope.
when checking that _ is a valid argument to a function of type
{M : Set → Set} {{r : R₂ M}} {A : Set} (x₁ : A) →
R₁.f (R₂.r₁ r) x₁ ≡ R₁.f (R₂.r₁ r) x₁
```

**Comment by nad** [comment]
The internal error for `internal-error₂` seems to be triggered starting with "[ fixed #1915 ] instance argument constraints not woken up properly" (85716a62899fd032073240f2915923b130adfb3a). However, I doubt that this commit introduced the bug.

**Comment by UlfNorell** [comment]
Seems to be a duplicate of #1974. After fixing that both examples give `No instance of type...` errors (as expected?).
